src/components/ChatPanel.tsx

```diff
@@ -92,12 +92,20 @@ export function ChatPanel(props: ChatPanelProps) {
     res.catch((e) => (handleError ? handleError(e) : defaultHandleApiError(e)));
   }, [chat, props, messages, defaultHandleApiError, canSendMessage]);
 
-  const messagesRef = useRef<HTMLDivElement>(null);
+  const messagesRef = useRef<Array<HTMLDivElement | null>>([]);
+  const messagesContainer = useRef<HTMLDivElement>(null);
 
   // Scroll to the bottom of the chat when the messages change
   useEffect(() => {
-    messagesRef.current?.scroll({
-      top: messagesRef.current?.scrollHeight,
+    if (messagesRef?.current.length < 2) {
```

Comment:
what happen if the first message is somehow super long, would it scroll down automatically or stay at the top?

Reply:
good check. I tested it, long first message stays at the top. I noticed there was an issue where restarting the conversation would scroll to the bottom, but I added a line to fix that.